src/web/components/TeamMember/TeamMembersTable.tsx

```diff
@@ -28,15 +30,33 @@ function TeamMembersTableContent({
   onRemoveTeamMember,
   onUpdateTeamMember,
 }: TeamMembersTableProps) {
+  const { LoggedInUser } = useContext(CurrentUserContext);
+  const { participant } = useContext(ParticipantContext);
+  const userRolesForCurrentParticipant = LoggedInUser?.user?.participants?.find(
+    (p) => p.id === participant?.id
+  )?.currentUserRoleIds;
+
   const [showTeamMemberDialog, setShowTeamMemberDialog] = useState<boolean>(false);
+  const [showTeamMemberActions, setShowTeamMemberActions] = useState<boolean>(false);
 
   const onOpenChangeTeamMemberDialog = () => {
     setShowTeamMemberDialog(!showTeamMemberDialog);
   };
 
-  const { sortData } = useSortable<UserDTO>();
+  const { sortData } = useSortable<UserResponse>();
   const sortedTeamMembers = sortData(teamMembers);
 
+  useEffect(() => {
+    const isUserAdminOrSupport =
```

Comment:
We do this logic here and in `PortalHeader.tsx` - perhaps it could be extracted into a helper that receives a `UserWithParticipantRoles` and returns a boolean